archinstall/lib/interactions/disk_conf.py

```diff
@@ -139,7 +139,6 @@ def select_disk_config(
 
 				return disk.DiskLayoutConfiguration(
 					config_type=disk.DiskLayoutType.Pre_mount,
-					relative_mountpoint=path,
```

Comment:
I'm a bit confused, this is the path that was entered by the user as the pre-mount root path, how are now using the entered value?

Reply:
Just above that `path` is passed to `detect_pre_mounted_mods()` which requires a base mountpoint.

https://github.com/archlinux/archinstall/blob/b141609990fa4f7305443ee6ea6fe8796604c539/archinstall/lib/interactions/disk_conf.py#L137-L138

The _use a best-effort_ option by default will use `/mnt/archinstall` but the `DiskLayoutConfiguration` does not need `relative_mountpoint`. This lead me to think the `mountpoint` of each `PartitionModification` can be modified before the list of them goes to the `DiskLayoutConfiguration`.

https://github.com/archlinux/archinstall/blob/4274a9a5c02826b05a4c08f55646e23c3ea79c29/archinstall/lib/disk/device_handler.py#L597-L599

This fails when using pre-mounted BTRFS subvolumes. A slight change to the above code to account for subvolumes (not shown here but just a `mountpoint` conditional and a for loop over the subvolumes doing the same thing, updating the mountpoint to account for it being inside a chroot) should be enough. When I test the change I think is needed for subvolumes the root partition used by the boot loader has a PARTUUID that is None.